ui/shared/components/form/ReduxFormWrapper.jsx

```diff
@@ -319,14 +325,14 @@ const mapStateToProps = (state, ownProps) => ({
 })
 
 const mapDispatchToProps = (dispatch, ownProps) => {
-  return {
+  return (ownProps.onSubmitSucceeded ? null : {
```

Comment:
this should be unnecessary, since you just won't call these functions in the case with `onSubmitSucceeded`. Also, in theory, you should be able to support both a success callback and a modal, so I would take this out

Reply:
I might be wrong but I couldn't get the cancel button in `ButtonPanel` to work without this

Reply:
with `noModal` you shouldn't need this

Reply:
on submit/cancel: `TypeError: Cannot read property 'confirmOnClose' of undefined`

Reply:
That error is because `handleClose` is being called but it shouldn't be when `noModal` is passed. You need to figure out why that is happening and fix that issue, not get rid of the function entirely